enhancements/installer/assisted-installer.md

```diff
@@ -0,0 +1,349 @@
+  - "@deads2k"
+  - "@hexfusion"
+  - "@mhrivnak"
+approvers:
+  - "@crawford"
+  - "@abhinavdahiya"
+  - "@eparis"
+creation-date: 2020-06-09
+last-updated: 2020-06-10
+status: implementable
+see-also:
+  - "/enhancements/baremetal/minimise-baremetal-footprint.md"
+---
+
+# Assisted Installer
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+1. Where do the UI and assisted installer application run for
+   disconnected installations?
+
+   * It could run from a host booted with the live ISO. If that host
+     is the bootstrap host, then access to the UI and API would be
+     interrupted when the host reboots to join the cluster. If that
+     host is not part of the cluster initially then the minimum
+     hardware footprint for creating a cluster increases.
+   * It could run on a user's desktop system, either from the ISO,
+     container image, or as a standalone binary. This may assume the
+     user is either running Linux or is willing to reboot their
+     desktop into a live ISO.
+
+2. In a disconnected deployment, how do the agents and assistant
+   application find each other?
+
+   * We could use generic or cluster-specific MDNS broadcast. MDNS is
+     not available in all environments and does not support hosts
+     running on different L2 networks. That also implies the
+     connection direction would be different for disconnected than
+     connected, where the agent reaches up to the service and the
+     service may not be able to reach the agent.
+   * We could build a URL into the ISO. That implies a separate ISO
+     for every cluster *and* the user would have to know the IP or
+     hostname of the bootstrap host in advance of building the
+     cluster.
+   * We could ask the user to enter a URL or other similar information
+     on the console when each host boots the live ISO, directing that
+     host to the current bootstrap host.
+   * We could ask the user to enter the IPs of hosts into the UI they
+     use to drive the installation and have the installer connect to
+     the agent on those IPs to instruct them to join the cluster. That
+     means the connection direction would be different for
+     disconnected than connected, where the agent reaches up to the
+     service and the service may not be able to reach the agent.
+
+## Summary
+
+This enhancement describes changes in and around the installer to
+assist with deployment on user-provisioned infrastructure. The use
+cases are especially relevant for bare metal, but will be applicable
+to cloud users who are running the installer directly instead of using
+a front-end such as `cloud.redhat.com` or the UI provided by their cloud
+vendor.
```

Comment:
The main thrust of this enhancement I think is "boot some machines with an ISO, they show up in the installer UI, and then you can kick off an install"

I think most potential reviewers of this would be puzzled by this intro saying the concept is applicable to other platforms, because the "boot some machines with an ISO" piece doesn't seem like it would ever be relevant?

I guess you're saying that an installer UI might be relevant? Maybe it would be good to say the focus is on bare metal and be explicit about exactly which parts of this overlap with ideas that could be relevant to other platforms?

Reply:
I was trying to incorporate some of the stuff that @crawford has mentioned, but that vision isn't really clear to me. Maybe it's a distraction and I should cut it?

Reply:
I rephrased this bit to push the other platform part further out into the future.